File: app/src/main/java/com/example/myownframework/KOTLIN_ANDROID/Codes/java/Factorial.java
package com.example.myownframework.Kotlin_Android.Codes.java;

public class Factorial {

    // logic F(n) = F(1)*F(2)...F(n-1)*F(n)

    public static void main(String[] args) {
        System.out.println(factorial(7));
    }

    public static long factorial(long n) {
        if (n == 1)
            return 1;
        else
            return (n * factorial(n - 1));
    }
}

File: app/src/main/java/com/example/myownframework/KOTLIN_ANDROID/Codes/kotlin/Factorial.kt
package com.example.myownframework.Kotlin_Android.Codes.kotlin

fun main(){

     println( factorial(7));

    greet


}

fun factorial(n: Long): Long {
    return if (n == 1L) 1 else n * factorial(n - 1)
}

val greet = { println("i am lambda") }




